C++/MST_Kruskals.cpp

```diff
@@ -0,0 +1,79 @@
+#include<bits/stdc++.h>
+using namespace std;
+#define ll long long
+#define fast ios_base::sync_with_stdio(false);	cin.tie(NULL);	cout.tie(NULL);
+#define piii pair<ll,pair<ll,ll>> 
+#define pii pair<ll,ll>
+#define pb push_back
+#define INF 0x3f3f3f3f
+int n;
+
+vector<piii> edges;
+vector<pii> parent ((int)1e4+5); 
+vector<pii> g[(int)1e4+5];
+
+ll find(ll x)
+{
+	if(parent[x].first==x)
+		return x;
+	return parent[x].first=find(parent[x].first);
+}
+
+void merge(ll x, ll y)
+{
+	ll set_x=find(x);									//Union based on the fact that both x and y are present in different sets.
+	ll set_y=find(y);
+	if(parent[set_x].second<parent[set_y].second)
+		parent[set_x].first=set_y;
+	else if(parent[set_x].second>parent[set_y].second)
+		parent[set_y].first=set_x;
+	else

```

Comment:
You forgot to increase the size of the Disjoined sets merged into one

Reply:
I have declared parent vector on line 12 of the code. Therefore, I don't have to resize it.

Reply:
You do need to increase size of the set, as parent shows the label of teh set but deciding factor is size. For best and most optimised complexity you need to increase the size

Reply:
In order to perform union on DSU, there are many possible heuristics that can be used. Most popular are the following two approaches: 

- In the first approach we use the size of the trees as rank.

- In the second one, we use the depth of the tree (more precisely, the upper bound on the tree depth, because the depth will get smaller when applying path compression), **this is the one which I have used**

In both approaches the essence of the optimization is the same: we attach the tree with the lower rank to the one with the bigger rank.

@YashSaxena9 can you please link the reference from where you found out that union by size is superior to union by rank? I tried to search for it, but couldn't find it. Would be happy to find out if this is the case. 🙂